lua/tree-climber/highlight.lua

```diff
@@ -0,0 +1,50 @@
+-- Credit: This file has largely been adapted from `vim.highlight.on_yank`,
+-- defined in neovim/runtime/lua/vim/highlight.lua.
+
+local M = {}
+
+local ts_utils = require("nvim-treesitter.ts_utils")
+
+local node_ns = vim.api.nvim_create_namespace("tree_climber_hlnode")
+local node_timer
+
+function M.highlight_node(node, opts)
+  vim.validate {
+    node = { node, "userdata", true },
+    opts = {
```

Comment:
I think all this can be simplified to `opts = { opts, "table", true }`. Or am I wrong?